packages/theming/apple-theme/macos/AppleThemeModule.swift

```diff
@@ -0,0 +1,427 @@
+import Foundation
+import FluentUI
+
+@objc(MSFAppleThemeModule)
+class AppleThemeModule: RCTEventEmitter {
+
+	public enum Events: CaseIterable {
+		case AppleInterfaceThemeChanged
+		case AppleColorPreferencesChanged
+
+		var name: String {
+			switch self {
+			case .AppleInterfaceThemeChanged:
+				return "AppleInterfaceThemeChanged";
+			case .AppleColorPreferencesChanged:
+				return "AppleColorPreferencesChanged";
+			}
+		}
+	}
+
+	override var methodQueue: DispatchQueue {
+		get {
+			return DispatchQueue.main
+		}
+	}
+
+	static func themingModule() -> [AnyHashable : Any] {
+		return [
+			"colors": palette(),
+			"typography" : [
+				"sizes" : fontSizes(),
+				"weights" : fontWeights(),
+				"families" : fontFamilies(),
+				"variants" : fontVariants()
+			],
+		]
+	}
+
+	override class func requiresMainQueueSetup() -> Bool {
+		return true
+	}
+
+	@objc(getApplePartialThemeWithCallback:)
+	func getApplePartialTheme(callback: RCTResponseSenderBlock) {
+		callback([NSNull(), AppleThemeModule.themingModule()])
+	}
+	
+	// MARK: - Colors
+
+	static func palette() -> [AnyHashable : Any] {
+
+		// Lifted from FluentUI Button as they aren't exposed publicly.
+		let brandForegroundDisabled = NSColor(named: "ButtonColors/brandForegroundDisabled", bundle: FluentUIResources.resourceBundle)!
+		let brandBackgroundDisabled = NSColor(named: "ButtonColors/brandBackgroundDisabled", bundle: FluentUIResources.resourceBundle)!
+		let neutralInverted = NSColor(named: "ButtonColors/neutralInverted", bundle: FluentUIResources.resourceBundle)!
+
+		return [
+			/* PaletteBackgroundColors & PaletteTextColors */
+
+			"background" : RCTColorToHexString(NSColor.windowBackgroundColor.cgColor),
+			"bodyStandoutBackground" : RCTColorToHexString(NSColor.underPageBackgroundColor.cgColor),
+			"bodyFrameBackground" : RCTColorToHexString(NSColor.windowBackgroundColor.cgColor),
+			"bodyFrameDivider" : RCTColorToHexString(NSColor.separatorColor.cgColor),
+			"bodyText" : RCTColorToHexString(NSColor.textColor.cgColor),
+			"bodyTextChecked" : RCTColorToHexString(NSColor.selectedTextColor.cgColor),
+			"subText" : RCTColorToHexString(NSColor.placeholderTextColor.cgColor),
+			"bodyDivider" : RCTColorToHexString(NSColor.separatorColor.cgColor),
```

Comment:
Re: Bug that fetching colors doesn't work properly.

Even though I am calling this again every-time the system appearance changes (say from light to dark), reevaluating this code will still give the original appearances. 

Example:
Boot in Light Mode
Background: #ececec
Switch to Dark Mode 

EXPECTED:
Background: #323232

ACTUAL:
Background: #ececec

I dunno what I'm doing wrong here, tagging other apple people @HeyImChris @harrieshin in case they catch something obvious

Reply:
you are actually recreating the dictionary?

Reply:
Yep. I can set breakpoints inside here and verify it's getting hit everyone I switch light / dark

Reply:
Removed the code for now. I'll follow up to see if this is a valid apple bug

Reply:
Update: using KVO rather than listening to NSNotifications worked. I still listen to the ColorPreferencesChangedNotification to see when the controlAccentColor changes.

```
func startObserving() {
	kvoToken = NSApplication.shared.observe(\.effectiveAppearance) { (application, change) in
		guard self.bridge != nil else {
			return
		}
		// Do work
	}
}

func stopObserving() {
	kvoToken?.invalidate()

}
```